File: API/Features/ShipCrews/Dtos/ShipCrewReadDto.cs
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.ShipCrews {

    public class ShipCrewReadDto : IMetadataRead {

        //  PK
        public int Id { get; set; }
        // Fields
        public SimpleEntity Gender { get; set; }
        public SimpleEntity Nationality { get; set; }
        public SimpleEntity Ship { get; set; }
        public string Lastname { get; set; }
        public string Firstname { get; set; }
        public string Birthdate { get; set; }
        public bool IsActive { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }

    }

}

File: FrontEnd/src/app/features/shipCrews/classes/dtos/shipCrew-read-dto.ts
import { GenderAutoCompleteVM } from 'src/app/features/genders/classes/view-models/gender-autocomplete-vm'
import { Metadata } from 'src/app/shared/classes/metadata'
import { NationalityDropdownVM } from 'src/app/features/nationalities/classes/view-models/nationality-autocomplete-vm'
import { ShipAutoCompleteVM } from 'src/app/features/ships/classes/view-models/ship-autocomplete-vm'

export interface ShipCrewReadDto extends Metadata {

    // PK
    id: number
    // Object fields
    gender: GenderAutoCompleteVM
    nationality: NationalityDropdownVM
    ship: ShipAutoCompleteVM
    // Fields
    lastname: string
    firstname: string
    birthdate: string
    isActive: boolean

}
